src/Database/Expression/TupleComparison.php

```diff
@@ -51,6 +51,16 @@ public function __construct($fields, $values, array $types = [], string $conjunc
         $this->_operator = $conjunction;
     }
 
+    /**
+     * Returns the type to be used for casting the value to a database representation
+     *
+     * @return array<string|null>
+     */
+    public function getType(): array
```

Comment:
Otherwise `$this->_type` could just be directly used.

Reply:
Oh, I've overlooked that the expression isn't `@internal` anymore... for quite some time already it seems.

Well, I guess it's `4.next` then, I don't see how I'd access `$this->_type` from the outside world any other way @dereuromark

Reply:
@ADmad Since this isn't an Interface, but a concrete class that is on top of that quite internal (likely not to be customized on that level to have a naming conflict of relevance here), I guess we could make this bugfix land in master.

Reply:
Yes, we can make an exception in this case considering it's quite low level and is fixing a bug.

Reply:
I agree this is fine to add in a bugfix release. We're deep in the guts of the ORM and this isn't a new method userland code will be using often.